src/coreclr/scripts/genEventPipe.py

```diff
@@ -382,14 +400,22 @@ def generateWriteEventBody(template, providerName, eventName, runtimeFlavor):
         body = "    ep_write_event (EventPipeEvent" + \
             eventName + ", (uint8_t *)buffer, (uint32_t)offset, ActivityId, RelatedActivityId);\n"
 
+    header += "\n"
     footer = ""
-    if runtimeFlavor.coreclr:
-        footer = """
+    if emittedWriteToBuffer:
+        if runtimeFlavor.coreclr:
+            footer = """
     if (!fixedBuffer)
         delete[] buffer;
+
+"""
```

Comment:
Couldn't we skip emitting this at all if `fixedBuffer` is already known to be true?

Reply:
Yup, and in C (mono) code, we are passing it by pointer so both caller and calllee indicate the linkage (like C#'s ref/out..), but in C++ it's passed by reference so call-sites does not immediately tell what's going to happen with each argument..

ps - I personally prefer using pointers in both C and C++ to keep it explicit both on caller and callee, but I didn't make that change here. 😅